tests/cephfs/cephfs_nfs/test_cephfs_nfs_ha_deployment_with_non_default_port.py

```diff
@@ -0,0 +1,177 @@
+import datetime
+import json
+import random
+import secrets
+import string
+import time
+import traceback
+
+from ceph.ceph import CommandFailed
+from ceph.parallel import parallel
+from tests.cephfs.cephfs_utilsV1 import FsUtils as FsUtilsV1
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    CEPH-83575097 - Create an NFS HA service with 2 or more daemons on a non default port
+        and test if the ingress / haproxy /keepalived daemons are running and the failover happens as expected.
+    Procedure
+    Scenario 1 - Create default NFS HA Deployment.
+    1. Log into the Cephadm shell:
+        Example
+        [root@host01 ~]# cephadm shell
+    2. Create the NFS cluster with the --ingress flag using a non default port:
+        Syntax
+        ceph nfs cluster create CLUSTER-ID [PLACEMENT] [--port PORT_NUMBER] [--ingress --virtual-ip IP_ADDRESS]
+    3. Validate the Service is UP
+    4. Mount using the Virtual IP.
+    5. Create Export and write IO.
+    6. Perform a failover and validate the IOs
+    7. Cleanup
+    """
+    try:
+        tc = "CEPH-83575097"
+        log.info(f"Running cephfs {tc} test case")
+        config = kw["config"]
+        build = config.get("build", config.get("rhbuild"))
+        fs_util_v1 = FsUtilsV1(ceph_cluster)
+        clients = ceph_cluster.get_ceph_objects("client")
+        client1 = clients[0]
+        fs_util_v1.prepare_clients(clients, build)
+        fs_util_v1.auth_list(clients)
+        nfs_name = "cephfsnfs"
+        virtual_ip = "10.8.128.100"
```

Comment:
should the IP addr be always hard-coded?

Reply:
Yeah, This is the free IP I requested from the Octo Lab maintainer and will be dedicated to Virtual IP for running nfs-ha suites.
As NFS-HA can be automated only in bare-metal, we do not have any other option to fetch free IPs and allocate it.

Reply:
thanks for the info